.changeset/blue-pens-divide.md

```diff
@@ -0,0 +1,22 @@
+---
+'astro': patch
+---
+
+Fixes a bug in the rewrite logic, where Astro didn't correctly rewrite to website with `base`.
```

Comment:
@sarah11918 The fix is the other way around. The **fix** and correct behaviour is that `base` **must** be included in the URL.

```diff
- Astro.rewrite("/blog") // incorrect
+ Astro.rewrite("/corporate/blog") // correct
```

Reply:
Right, I think we're saying the same thing? the error was that base was included by Astro, so the user didn't have to write it. Now, after fixing, Astro no longer includes it, so the user must?

Or, are you saying if you had a base configured before, it was NEVER possible to get `Astro.rewrite` to work? That's why I had asked before, what if people *did* get it to work and now need to change because the "fix" breaks their project.

In other words, is it:

> Fixes a bug in the logic of `Astro.rewrite()` which did not work if you had a `base` configured.

> You can now use the `rewrite()` function on a project that has `base` configured. To do so, you must prepend the base to your existing rewrite URL:

??

Reply:
> but when I read the suggestion, I understood that the **users** were led to add base in their URL.

Just checking, because in line 30 that is exactly what we're telling them to do, right? Users *do* need to ensure that base is in their URL?

Reply:
@ematipico  I tried again! See if that is clearer!

Reply:
That's way better, thank you Sarah